Dockerfile.windows

```diff
@@ -128,55 +154,107 @@ SHELL ["powershell", "-command"]
 #  - GO_VERSION must be consistent with 'Dockerfile' used by Linux.
 #  - FROM_DOCKERFILE is used for detection of building within a container.
 ENV GO_VERSION=1.7.3 `
-    GIT_LOCATION=https://github.com/git-for-windows/git/releases/download/v2.10.1.windows.1/Git-2.10.1-64-bit.exe `
+    GIT_VERSION=2.10.2 `
     GOPATH=C:\go `
     FROM_DOCKERFILE=1
 
-WORKDIR C:\
-
 RUN `
-  setx /M Path $($Env:PATH+';C:\gcc\bin;C:\go\bin'); `
-  `
   $ErrorActionPreference = 'Stop'; `
+  $ProgressPreference = 'SilentlyContinue'; `
+  `
+  Function Test-Nano() { `
+    $EditionId = (Get-ItemProperty -Path 'HKLM:\SOFTWARE\Microsoft\Windows NT\CurrentVersion' -Name 'EditionID').EditionId; `
+	return $True; `
+    return (($EditionId -eq 'ServerStandardNano') -or ($EditionId -eq 'ServerDataCenterNano') -or ($EditionId -eq 'NanoServer')); `
+  }`
+  `
   Function Download-File([string] $source, [string] $target) { `
-    $wc = New-Object net.webclient; $wc.Downloadfile($source, $target) `
+      $webClient = New-Object System.Net.WebClient; `
+      $webClient.DownloadFile($source, $target); `
+	return; `
+    if (Test-Nano) { `
```

Comment:
Currently yes. It works on nano right up to the point I mentioned in the notes section.